From 09e7145f243ef4f2c00cd57a3e3458cbc0cf0dcc Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 16 Aug 2016 14:32:19 +0300
Subject: [PATCH 0421/1345] net: mvpp2x: set buffer pool size as boot
 configurable parameter

commit  8ae529089ad53192ca05d12a11bc5acfda2070cd from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- add module_params: short_pool, long_pool, jumbo_pool

Change-Id: I5e63de1c91a3a14189c8acacd51b465b87e1f3b8
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31979
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |   42 +++++++++++---------
 1 file changed, 24 insertions(+), 18 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index b6bb098..634d655 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -89,6 +89,21 @@
 
 u32 debug_param;
 
+struct mv_pp2x_pool_attributes mv_pp2x_pools[] = {
+	{
+		.description =  "short", /* pkt_size=MVPP2_BM_SHORT_PKT_SIZE */
+		.buf_num     =  MVPP2_BM_SHORT_BUF_NUM,
+	},
+	{
+		.description =  "long", /* pkt_size=MVPP2_BM_LONG_PKT_SIZE */
+		.buf_num     =  MVPP2_BM_LONG_BUF_NUM,
+	},
+	{
+		.description =	"jumbo", /* pkt_size=MVPP2_BM_JUMBO_PKT_SIZE */
+		.buf_num     =  MVPP2_BM_JUMBO_BUF_NUM,
+	}
+};
+
 module_param_named(num_cos_queues, mv_pp2x_num_cos_queues, byte, S_IRUGO);
 MODULE_PARM_DESC(num_cos_queues, "Set number of cos_queues (1-8), def=4");
 
@@ -129,6 +144,15 @@
 MODULE_PARM_DESC(debug_param,
 	"Ad-hoc parameter, which can be used for various debug operations.");
 
+module_param_named(short_pool, mv_pp2x_pools[MVPP2_BM_SWF_SHORT_POOL].buf_num, uint, S_IRUGO);
+MODULE_PARM_DESC(short_pool, "Short pool size (0-8192), def=2048");
+
+module_param_named(long_pool, mv_pp2x_pools[MVPP2_BM_SWF_LONG_POOL].buf_num, uint, S_IRUGO);
+MODULE_PARM_DESC(long_pool, "Long pool size (0-8192), def=1024");
+
+module_param_named(jumbo_pool, mv_pp2x_pools[MVPP2_BM_SWF_JUMBO_POOL].buf_num, uint, S_IRUGO);
+MODULE_PARM_DESC(jumbo_pool, "Jumbo pool size (0-8192), def=512");
+
 /*TODO:  Below module_params will not go to ML. Created for testing. */
 
 module_param(port_cpu_bind_map, uint, S_IRUGO);
@@ -157,24 +181,6 @@ void set_device_base_address(struct net_device *dev)
 /* Number of TXQs used by single port */
 static int mv_pp2x_txq_number;
 
-struct mv_pp2x_pool_attributes mv_pp2x_pools[] = {
-	{
-		.description =  "short",
-		/*.pkt_size    =	MVPP2_BM_SHORT_PKT_SIZE, */
-		.buf_num     =  MVPP2_BM_SHORT_BUF_NUM,
-	},
-	{
-		.description =  "long",
-		/*.pkt_size    =	MVPP2_BM_LONG_PKT_SIZE,*/
-		.buf_num     =  MVPP2_BM_LONG_BUF_NUM,
-	},
-	{
-		.description =	"jumbo",
-		/*.pkt_size    =	MVPP2_BM_JUMBO_PKT_SIZE, */
-		.buf_num     =  MVPP2_BM_JUMBO_BUF_NUM,
-	}
-};
-
 static inline int mv_pp2x_txq_count(struct mv_pp2x_txq_pcpu *txq_pcpu)
 {
 
-- 
1.7.9.5

